client/app/src/main/java/com/microsoft/research/karya/utils/FileUtils.kt

```diff
@@ -145,4 +145,22 @@ object FileUtils {
     // Fill to 32 chars
     return "%32s".format(output).replace(' ', '0')
   }
+
+  fun createDirectory(dirPath: String): String {
+
+    val dir = File(dirPath)
+
+    var success = true
+    if (!dir.exists()) {
+      success = dir.mkdirs()
+    }
+
+    if (success) {
+      return dirPath
+    } else {
+      throw FileNotFoundException()
+    }
```

Comment:
You can probably rewrite it to something like this

```suggestion
    if (dir.exists()) return dirPath
    if (!dir.exists() && dir.mkdirs()) return dirPath
    throw FileNotFoundException()
```

You can probably combine the two if conditions too with something like this `if (dir.exists() || (!dir.exists() && dir.mkdirs()))` but I feel it's more clean with having those two separate.